### PR TITLE
Fix capitalization

### PR DESCRIPTION
**Pull request by klaplong**
Fixed https://github.com/nvie/pip-tools/issues/431

### CODE REVIEW

**Comment by barrywhart** [comment]
Any idea why the build is failing? I can't access the Travis log.

**Comment by davidovich** [comment]
The project has moved to the jazzband organization and it seems that past builds have not followed from nvie. 

Pushing another commit (or amending the last commit) should trigger a build in the new organization.

**Comment by rpkilby** [comment]
If you change the url to jazzband from nvie, it should show the build.

**Comment by davidovich** [comment]
https://travis-ci.org/jazzband/pip-tools/builds/199895230

Note however that the "This branch has no conflicts with the base branch" message is flawed here. Rebasing on master should help.

**Comment by barrywhart** [comment]
This looks like a good fix for the capitalization behavior. In the original issue (https://github.com/jazzband/pip-tools/issues/431), I had raised some questions about some possible cache-related weirdness. Should I create another GitHub issue for that? Either way (new issue or no), I will need to go back to try and reproduce the issue.

**Comment by klaplong** [comment]
Yes I have read a lot of code, but I could not really find where the caching actually takes place. This seems to be part of another package.

File: tests/test_writer.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def test_format_requirement_annotation_case_sensitive(from_line, writer):
     assert (writer._format_requirement(ireq,
                                        reverse_dependencies,
                                        primary_packages=[]) ==
-            'Test==1.2                 ' + comment('# via xyz'))
```
Comment (davidovich) [inline tests/test_writer.py]: This test is made to explicitly validate that case is NOT changed. (test is named test_format_requirement_annotation_case_sensitive). Are we sure this is ok ?

Comment (klaplong) [inline tests/test_writer.py]: Case _sensitivity_ is not the same as _unchanged_ case. On the other hand, neither pip, nor pypi, nor pip-tools (as it uses pip itself) are case sensitive. Maybe it's better to either just get rid of this test, or rename it, or even write another one?

Comment (davidovich) [inline tests/test_writer.py]: Yes that makes sense. 

Going forward, I think adding and maintaining tests is really important, particularly in this new "distributed" maintainership model (jazzband) that pip-tools is using. I know it is more work, but it is better to spend time locally than distribute the debugging effort on the internet :-). 

Maybe this should be included in the 1.8.1 release I am preparing.  A release candidate 1 is out already.

**Comment by klaplong** [comment]
I do have a feeling we should document this case-insensitivity somewhere, although I have no idea where.

**Comment by davidovich** [comment]
Thanks @klaplong !

**Pull request merged**